src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppAppRunner/Generated/Recipe.cs

```diff
@@ -44,20 +46,55 @@ public Recipe(Construct scope, IRecipeProps<Configuration> props)
             ConfigureAppRunnerService(props);
         }
 
+        private void ConfigureVpc(Configuration settings)
+        {
+            if (settings.VPCConnector.UseVPCConnector)
+            {
+                if (settings.VPCConnector.CreateNew)
+                {
+                    if (settings.VPCConnector.CreateNewVpc)
+                    {
+                        AppVpc = new Vpc(this, nameof(AppVpc), InvokeCustomizeCDKPropsEvent(nameof(AppVpc), this, new VpcProps
+                        {
+                            MaxAzs = 2
+                        }));
+                    }
+                }
+            }
+        }
+
         private void ConfigureVPCConnector(Configuration settings)
         {
             if (settings.VPCConnector.CreateNew)
             {
-                if (settings.VPCConnector.Subnets.Count == 0)
-                    throw new InvalidOrMissingConfigurationException("The provided list of Subnets is null or empty.");
+                if (settings.VPCConnector.CreateNewVpc)
+                {
+                    ConfigureVpc(settings);
+
+                    if (AppVpc == null)
+                        throw new InvalidOperationException($"{nameof(AppVpc)} has not been set.");
+
+                    VPCConnector = new CfnVpcConnector(this, nameof(VPCConnector), InvokeCustomizeCDKPropsEvent(nameof(VPCConnector), this, new CfnVpcConnectorProps
+                    {
+                        Subnets = AppVpc.PrivateSubnets.Select(x => x.SubnetId).ToArray(),
 
-                VPCConnector = new CfnVpcConnector(this, nameof(VPCConnector), InvokeCustomizeCDKPropsEvent(nameof(VPCConnector), this, new CfnVpcConnectorProps
+                        // the properties below are optional
+                        SecurityGroups = new string[] { AppVpc.VpcDefaultSecurityGroup }
+                    }));
```

Comment:
I noticed when I ran this that sub parts of the new VPC would be created after the VPC connector and while the App Runner service is being created. This could get us into odd timing issues if the VPC doesn't complete before App Runner gives up on health checks. If you add the following line right after creating the `VPCConnector` that will ensure the VPC connector is not created till the VPC is completely created.

```
VPCConnector.Node.AddDependency(AppVpc);
```

Reply:
Done